File: Python/canConstruct.py
def canConstruct(ransomNote : str, magazine : str)-> bool:
  map = {}
  for char in magazine:
    if char in map:
      map[char] += 1
    else:
      map[char] = 1
    
  for char in ransomNote:
    if char in map and map[char] > 0:
      map[char] -= 1
    else:    
      return False
  
  return True


ransomNote = "aabb"
magazine = "aab"
answer = canConstruct(ransomNote, magazine)
print(answer)

File: Java/canConstruct.java
package Leetcodes;
import java.util.HashMap;
import java.util.Map;

public class canConstruct {
    public static boolean canConstruct(String magazine, String ransomeNote){
        Map<Character, Integer> map = new HashMap<>();
        for(Character c : magazine.toCharArray()){
            map.put(c, map.getOrDefault(c, 0) + 1);
        }
        for(Character c : ransomeNote.toCharArray()){
            if(map.containsKey(c) && map.get(c) > 0){
                map.put(c, map.get(c)-1);
            }else{
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        String magazine = "aab";
        String ransomeNote = "aa";
        System.out.println(canConstruct(magazine, ransomeNote));
    }
}
